File: src/main/java/com/julien/juge/khast/api/entity/PostEntity.java
package com.julien.juge.khast.api.entity;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Data
@Document(collection = "Post")
public class PostEntity {

    @Id
    private String id;

    @CreatedDate
    private Date date;


}

File: src/main/java/com/julien/juge/khast/api/entity/PostEntity.kt
package com.julien.juge.khast.api.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document(collection = "Post")
class PostEntity {

    constructor(date: Date?, event: EventEntity?, text: TextEntity?)

    @Id
    var id: String? = null
        set(id) {
            field = this.id
        }

    var date: Date? = null
        set(date) {
            field = this.date
        }

    var event: EventEntity? = null
        set(event) {
            field = this.event
        }

    var text: TextEntity? = null
        set(text) {
            field = this.text
        }
}
